docs/contract.yaml

```diff
@@ -0,0 +1,158 @@
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Success"
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"  
+
+  /operations/overlay:
+    post:
+      description: |
+        the overlay endpoint takes a spec and an overlay file (which is a partial spec)
+        and performs a specialized union where the overlay takes priority. In case of a collision, a
+        conflict will not be reported and the value from the overlay will be taken.
+      operationId: overlay
+      requestBody:
+        description: data needed for the overlay operation
+        required: true
+        content: 
+          application/json:
+            schema: 
+              "$ref": "#/components/schemas/OverlayRequest"
+      responses:
+        '200':
+          description: operation was successful, returns the resultant spec 
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Success"   
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"                
+components:
+  schemas:
+    MergeRequest:
+      description: holds data required for the union operation
+      type: object
+      required:
+        - spec1
+        - spec2
+      properties:
+        spec1:
+          description: an OpenAPI specification
+          type: string
+        spec2:
+          description: an OpenAPI specification
+          type: string
+        defaults:
+          description: an OpenAPI specification fragment which contains metadata about the merged spec
+          type: string
+        conflictResolutions:
+          description: an array of MergeConflict objects which specify resolutions to conflicting keypaths
+          type: array
+          items:
+            "$ref": "#/components/schemas/MergeConflict"
+    
+    OverlayRequest:
+      description: holds data required for the overlay operation
+      type: object
+      required:
+        - spec
+        - overlay
+      properties:
+        spec:
+          description: an OpenAPI specification
+          type: string
+        overlay:
+          description: an OpenAPI specification fragment which contains metadata about the merged spec
+          type: string
+
+    MergeConflict:
+      description: an individual conflict
+      type: object
+      properties:
+        keypath:
+          type: string
+          example: paths:/pet:post:summary
+        option1:
+          type: string
+          example: Post a single pet
+        option2:
+          type: string
+          example: Post several pets
+        resolvedValue:
+          type: string
+          example: Post a single pet
+    
+    Success:
```

Comment:
It might create confusion to call this "Success" when it includes a status field that could be either "success", "conflicts", or "operation error".

Looking to https://google.aip.dev/136 for patterns, I think it would be good to call this a "Response" object, and since it isn't exclusively a "MergeResponse" or an "OverlayResponse", I'd suggest calling it "OperationResponse".

Reply:
Agreed